File: src/types/backupTrash.ts
import type { Project } from './project';
import type { ManuscriptNode, DocumentContent } from './manuscript';
import type { Character, CharacterRelationship } from './character';
import type { Location } from './location';
import type { WorldbuildingEntry } from './worldbuilding';
import type { TimelineEvent } from './timeline';
import type { Note } from './note';
import type { Tag } from './tag';
import type { WritingGoal, WritingSession } from './phase5';

export interface EntityTag {
  entity_type: string;
  entity_id: string;
  tag_id: string;
}

export interface TrashItem {
  id: string;
  project_id: string;
  entity_type: 'manuscript' | 'character' | 'location' | 'note';
  entity_id: string;
  title: string;
  deleted_at: string;
}

export interface BackupFileInfo {
  fileName: string;
  filePath: string;
  fileSizeBytes: number;
  createdAt: string;
}

export interface BackupResult {
  fileName: string;
  filePath?: string | null;
  contentJson: string;
}

export interface ProjectBackupBundle {
  version: string;
  exportedAt: string;
  project: Project;
  nodes: ManuscriptNode[];
  documents: Record<string, DocumentContent>;
  characters: Character[];
  relationships: CharacterRelationship[];
  locations: Location[];
  worldbuilding: WorldbuildingEntry[];
  timeline: TimelineEvent[];
  notes: Note[];
  tags: Tag[];
  entityTags: EntityTag[];
  writingGoals: WritingGoal[];
  writingSessions: WritingSession[];
  settings: Record<string, string>;
}

File: src-tauri/src/models/backup_trash.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

use crate::models::{
    Character, CharacterRelationship, DocumentContent, Location, ManuscriptNode,
    Note, Project, TimelineEvent, WorldbuildingEntry, WritingGoal, WritingSession,
};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TrashItem {
    pub id: String,
    pub project_id: String,
    pub entity_type: String, // "manuscript" | "character" | "location" | "note"
    pub entity_id: String,
    pub title: String,
    pub deleted_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BackupFileInfo {
    pub file_name: String,
    pub file_path: String,
    pub file_size_bytes: u64,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BackupResult {
    pub file_name: String,
    pub file_path: Option<String>,
    pub content_json: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProjectBackupBundle {
    pub version: String,
    pub exported_at: String,
    pub project: Project,
    pub nodes: Vec<ManuscriptNode>,
    pub documents: HashMap<String, DocumentContent>,
    pub characters: Vec<Character>,
    pub relationships: Vec<CharacterRelationship>,
    pub locations: Vec<Location>,
    pub worldbuilding: Vec<WorldbuildingEntry>,
    pub timeline: Vec<TimelineEvent>,
    pub notes: Vec<Note>,
    pub writing_goals: Vec<WritingGoal>,
    pub writing_sessions: Vec<WritingSession>,
    pub settings: HashMap<String, String>,
}
